File: Exercism/typescript/matrix/matrix.ts
const zip = (...arrays: number[][]): number[][] => {
  const iters = arrays.map((array) => array[Symbol.iterator]())
  let items = iters.map((iter) => iter.next())
  const columns = []
  while (!items.some((item) => item.done)) {
    columns.push(items.map((item) => item.value))
    items = iters.map((iter) => iter.next())
  }
  return columns
}

class Matrix {
  rows: number[][]
  columns: number[][]

  constructor (matrix: string) {
    this.rows = matrix
      .split('\n')
      .map((row) => row.split(' ').map(Number))
    this.columns = zip(...this.rows)
  }
}

export default Matrix

File: Exercism/python/matrix/matrix.py
class Matrix(object):
    def __init__(self, matrix_string):
        self._matrix = [[int(cell) for cell in row.split(' ')]
                        for row in matrix_string.split('\n')]

    @property
    def matrix(self):
        return [row[:] for row in self._matrix]

    def row(self, index):
        return [cell for cell in self.matrix[index - 1]]

    def column(self, index):
        return [row[index - 1] for row in self.matrix]
